datafusion/physical-expr/src/window/aggregate.rs

```diff
@@ -155,8 +155,7 @@ impl WindowExpr for PlainAggregateWindowExpr {
     }
 
     fn uses_bounded_memory(&self) -> bool {
-        self.aggregate.supports_bounded_execution()
-            && !self.window_frame.end_bound.is_unbounded()
+        !self.window_frame.end_bound.is_unbounded()
```

Comment:
After thinking through this logic. Actually, as long as `end_bound` is not bounded (not `UNBOUNDED FOLLOWING` such as in the form `N FOLLOWING`). We can produce results without waiting for the whole data to come (If accumulator do not support `retract_batch` method. We wouldn't be able to run queries in the form `M PRECEDING and N FOLLOWING`, in this case we will give an error anyway.). Hence here we do not need to check for `self.aggregate.supports_bounded_execution()` (`acc.supports_retract_batch()` method with the new API.)

Reply:
This makes sense to me